pandas/core/indexes/base.py

```diff
@@ -5390,6 +5390,89 @@ def get_indexer_for(self, target) -> np.ndarray:
         indexer, _ = self.get_indexer_non_unique(target)
         return indexer
 
+    def _get_indexer_strict(self, key, axis_name: str_t) -> tuple[Index, np.ndarray]:
+        """
+        Analogue to get_indexer that raises if any elements are missing.
+        """
+        keyarr = key
+        if not isinstance(keyarr, Index):
+            keyarr = com.asarray_tuplesafe(keyarr)
+
+        if self._index_as_unique:
+            indexer = self.get_indexer_for(keyarr)
+            keyarr = self.reindex(keyarr)[0]
+        else:
+            keyarr, indexer, new_indexer = self._reindex_non_unique(keyarr)
+
+        self._raise_if_missing(keyarr, indexer, axis_name)
+
+        if (
```

Comment:
is_extension_array_dtype?

Reply:
also dt64 and td64.  im pretty sure we should actually be doing this (.take on L5416) unconditionally, but that breaks a few tests which i think are wrong, so thats not ready yet